src/client.cpp

```diff
@@ -473,7 +473,7 @@ void UpdatePort(unsigned int port, bool connected, const game_controller* contro
 
   const unsigned int device = CInputManager::Get().GetDevice(port);
 
-  if (CLIENT)
+  if (CLIENT && (int)port >= 0)
```

Comment:
We use the port number in a few places as map index (such as m_ports, m_devices). If we want to restrict ports to be unsigned, we need to store the instances related to mouse separately and also access them in a different way. I think making it (signed) int everywhere would be simpler. this line here could be `if (CLIENT && port != GAME_INPUT_PORT_MOUSE)`